numba/tests/test_array_methods.py

```diff
@@ -1329,6 +1335,45 @@ def test_imag(self):
         z = x + 1j*y
         np.testing.assert_equal(pyfunc(z), cfunc(z))
 
+    def test_clip(self):
+        # TODO: scalars are not tested (issue #3469)
+        for a in (np.linspace(-10, 10, 101),
+                  np.linspace(-10, 10, 40).reshape(5, 2, 4)):
+            for pyfunc in [np_clip, array_clip]:
+                cfunc = jit(nopython=True)(pyfunc)
+
+                with self.assertRaises(ValueError):
+                    pyfunc(a, None, None)
+                with self.assertRaises(ValueError):
```

Comment:
Assert the correct error string?

Reply:
I've added an assertion for the correct error string. I removed the check of the raise for the pyfunc, as it wasn't really testing anything to do with Numba.